ssl/tls13_enc.c

```diff
@@ -764,12 +774,53 @@ int tls13_change_cipher_state(SSL *s, int which)
         s->statem.enc_write_state = ENC_WRITE_STATE_WRITE_PLAIN_ALERTS;
     else
         s->statem.enc_write_state = ENC_WRITE_STATE_VALID;
+#ifndef OPENSSL_NO_KTLS
+# if defined(OPENSSL_KTLS_TLS13)
+#  ifndef __FreeBSD__
+    if (!(which & SSL3_CC_WRITE)
+        || ((which & SSL3_CC_WRITE) && (s->mode & SSL_MODE_NO_KTLS_TX)))
+        goto skip_ktls;
+
+    /* ktls supports only the maximum fragment size */
+    if (ssl_get_max_send_fragment(s) != SSL3_RT_MAX_PLAIN_LENGTH)
```

Comment:
Ah!! We never enter KTLS if max_fragment_length is in use. Probably we also need to consider the `record_padding_cb` here. Strictly speaking the `record_padding_cb` could be set at any time (even after the handshake). Probably we should fail to set it if KTLS is in use?? (Would need to update docs in that case)

Reply:
Additional check is added in configuration.
And also added checks to SSL_set_record_padding_callback. Checks are not needed in SSL_CTX_set_record_padding_callback because this value is not propagated into already established SSLs.